### PR TITLE
[Dashboard] Display duration in milliseconds if under 1 second.

### LINKED ISSUE #49010: [Dashboard] For durations less than 1s, show ms instead
### Description

Screenshot 2024-11-26 at 8 36 18 AM

Duration of "0s" is not very useful to show.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by coltwood93**
## Why are these changes needed?

On the Ray Dashboard, a duration less than 1 second is displayed as "0s", which is not accurate.

## Related issue number

Closes #49010 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dashboard/client/src/common/DurationText/DurationText.tsx [hunk 1]
```diff
@@ -27,7 +27,9 @@ export const DurationText = ({ startTime, endTime }: DurationTextProps) => {
 
   let durationText: string;
   let refreshInterval = 1000;
-  if (duration.asMinutes() < 1) {
+  if (duration.asSeconds() < 1) {
+    durationText = duration.format("SSS[ms]");
+  } else if (duration.asMinutes() < 1) {
```
Comment (alanwguo) [inline python/ray/dashboard/client/src/common/DurationText/DurationText.tsx:32]: For numbers less than 1 minute, can we include milliseconds as well?

**Review by alanwguo — COMMENTED** [review]
Thanks for picking this up!

File: python/ray/dashboard/client/src/common/DurationText/DurationText.tsx (hunk 1 above)
Comment (alanwguo) [reply python/ray/dashboard/client/src/common/DurationText/DurationText.tsx:32]: also, can you add a test case in `DurationText.component.test.tsx`?

Comment (coltwood93) [reply python/ray/dashboard/client/src/common/DurationText/DurationText.tsx:32]: > For numbers less than 1 minute, can we include milliseconds as well?

Sure, I can add that! I'll add it momentarily.

> also, can you add a test case in DurationText.component.test.tsx?

Yes, I'll take a look and see what I can come up with. Thanks for the review!

**Comment by coltwood93** [comment]
I mistakenly committed without signing, so I'm trying to figure out how to correct that...

**Comment by coltwood93** [comment]
I think I just blew up this PR. Going to close and open a fresh one to resolve issue #49010

**Pull request closed without merging**